### DISCUSSION TITLE
help on build wpewebkit+cog on yocto zeus (imx6q)

### DISCUSSION THREAD

**Discussion by supercalhotas** [Igalia/meta-webkit#274] [General]
Hi guys,

I'm trying to integrate wpewebkit+cog on imy image and followed up your guide on imx6 for the fullscreen app, but i'm not making any progress so far:

**my conf/local.conf:**
IMAGE_INSTALL_append = " gstreamer1.0-plugins-imx gstreamer1.0-plugins-imx-meta"
IMAGE_INSTALL_append = " wpewebkit cog"
PREFERRED_PROVIDER_virtual/wpebackend = "wpebackend-rdk"
PACKAGECONFIG_pn-wpebackend-rdk = "imx6"

**the error i'm stucked in:**
z0rd@a138ef353c0d:/opt/ts-bsp/build$ bitbake wpewebkit
Loading cache: 100% |#########################################################################################################################################################| Time: 0:00:00Loaded 3845 entries from dependency cache.
NOTE: Resolving any missing task queue dependencies
ERROR: Nothing PROVIDES 'wpebackend-fdo' (but /opt/ts-bsp/sources/meta-webkit/recipes-browser/wpewebkit/wpewebkit_2.28.1.bb DEPENDS on or otherwise requires it)
wpebackend-fdo was skipped: PREFERRED_PROVIDER_virtual/wpebackend set to wpebackend-rdk, not wpebackend-fdo
wpebackend-fdo was skipped: PREFERRED_PROVIDER_virtual/wpebackend set to wpebackend-rdk, not wpebackend-fdo
wpebackend-fdo was skipped: PREFERRED_PROVIDER_virtual/wpebackend set to wpebackend-rdk, not wpebackend-fdo
wpebackend-fdo was skipped: PREFERRED_PROVIDER_virtual/wpebackend set to wpebackend-rdk, not wpebackend-fdo
wpebackend-fdo was skipped: PREFERRED_PROVIDER_virtual/wpebackend set to wpebackend-rdk, not wpebackend-fdo
wpebackend-fdo was skipped: PREFERRED_PROVIDER_virtual/wpebackend set to wpebackend-rdk, not wpebackend-fdo
wpebackend-fdo was skipped: PREFERRED_PROVIDER_virtual/wpebackend set to wpebackend-rdk, not wpebackend-fdo
ERROR: Required build target 'wpewebkit' has no buildable providers.
Missing or unbuildable dependency chain was: ['wpewebkit', 'wpebackend-fdo']

Summary: There were 2 ERROR messages shown, returning a non-zero exit code

Is there something i should change on my local config? (zeus branch)

**Comment by philn** [comment]
Do you have the qt layer enabled? This commit might be messing up your setup: https://github.com/Igalia/meta-webkit/commit/d7c92f3533c9a25436a4373b022d3b9ae7740324 Try to revert?

**Reply by GobGeeklord** [reply]
Hi @philn, 

Thanks for your help! First of all, my target is to run some wep apps that have embedded videos on it.
Yup, i checked it out before and i was able to build the image but unfortunately i am not able to run cog directly on framebuffer.
When i run cog, it doesn't output nothing to my HDMI screen:
https://pastebin.com/3Y6dLCRq
But i can play video using gst-play1.0.

I've tried to build the image with Weston on fbdev, and the cog works inside Weston, but the video doesn't work, is always black. (for example: https://people.igalia.com/clopez/wkbug/video/simplevideo.html)

my local.conf:
IMAGE_INSTALL_append = " gstreamer1.0-plugins-imx gstreamer1.0-plugins-imx-meta wpewebkit cog"
DISTRO_FEATURES_append = " opengl wayland"
PREFERRED_PROVIDER_virtual/wpebackend = "wpebackend-rdk"
PACKAGECONFIG_pn-wpebackend-rdk  = "imx6"

PREFERRED_VERSION_openssh = "7.8p1+git"
PREFERRED_VERSION_openssl = "1.1.1b"
PREFERRED_VERSION_openssl-native = "1.1.1b"
PREFERRED_VERSION_nodejs = "8.12.0"

Am i missing something?

Regards

**Reply by philn** [reply]
I haven't used the RDK backend at all, even less on iMX, so can't help much with that.

One approach we often recommend, if you don't use a wayland compositor you can still use WPEBackend-FDO and then enable the DRM platform backend in Cog. That will enable KMS/DRM full-screen rendering, which is often enough for kiosk-like apps. 

If you have kmscube working on your device, then Cog/DRM should also work.

**Reply by GobGeeklord** [reply]
Hi @philn ,

Oh, i see!
In the configuration you're saying, there is video aceleration available?
How do i enable DRM module to be built/available in my image? (i saw in the release notes of cog 0.9.1 that module is built by default, so i'm assuming that in 0.8.X it isn't?)

IMAGE_INSTALL_append = " gstreamer1.0-plugins-imx gstreamer1.0-plugins-imx-meta wpewebkit cog"
DISTRO_FEATURES_append = " opengl"
PREFERRED_PROVIDER_virtual/wpebackend = "wpebackend-fdo"

Thanks in advance!

**Reply by philn** [reply]
the DRM option is not available in the Zeus branch. You can try to update to a new version back ported from main though.

**Comment by philn** [comment]
Sorry for the late reply btw. New discussions don't trigger notifications, some bad UX there...
